src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectExtensions.cs

```diff
@@ -18,10 +18,15 @@ public static class OpenIdConnectExtensions
         /// <param name="app">The application builder</param>
         /// <param name="options">Options which control the processing of the OpenIdConnect protocol and token validation.</param>
         /// <returns>The application builder</returns>
-        public static IApplicationBuilder UseOpenIdConnectAuthentication(this IApplicationBuilder app, Action<OpenIdConnectOptions> configureOptions = null)
+        public static IApplicationBuilder UseOpenIdConnectAuthentication(this IApplicationBuilder app, Action<OpenIdConnectOptions> configureOptions)
         {
-            return app.UseMiddleware<OpenIdConnectMiddleware>(
-                 new ConfigureOptions<OpenIdConnectOptions>(configureOptions ?? (o => { })));
+
+            var options = new OpenIdConnectOptions();
+            if (configureOptions != null)
```

Comment:
This will be the new pattern with NotNull dead, might as well just allow it and no-op